spec/views/fines/index.html.erb_spec.rb

```diff
@@ -3,48 +3,114 @@
+    context 'when the patron has no fines' do
+      let(:patron) do
+        # create a patron with empty array for fines
+        instance_double(Symphony::Patron, barcode: '1', fines: [], can_pay_fines?: true, requests: [], checkouts: [],
+                                          remaining_checkouts: nil, barred?: false, status: 'OK', group?: false)
+      end
+
+      it 'does not show the shared computer payment alert' do
+        render
+
+        expect(rendered).not_to have_text('Shared computer users: Due to computer security risks, you should not use a shared computer to make a fine payment.') # rubocop:disable Layout/LineLength
+      end
     end
   end
 
-  context 'when the patron has no fines' do
+  context 'with a FOLIO Fine' do
+    let(:fine) do
+      instance_double(Folio::Fine,
+                      owed: 3,
+                      status: 'A',
+                      sequence: '1',
+                      nice_status: 'Damaged',
+                      bib?: true,
+                      key: 'abc',
+                      bill_date: Date.new,
+                      fee: 5,
+                      library:
+                      'Best Lib',
+                      barcode: '12345',
+                      author: 'Author 1',
+                      title: 'Title',
+                      shelf_key: 'AB 1234',
+                      call_number: 'AB 1234',
+                      catkey: '12345')
+    end
+    let(:fines) { [fine] }
+    let(:checkouts) do
+      [Folio::Checkout.new(
+        { 'id' => '31d15973-acb6-4a12-92c7-5e2d5f2470ed',
+          'item' =>
+           { 'title' =>
+             'Mental growth during the first three years' },
+          'overdue' => true,
+          'details' =>
+           { 'feesAndFines' => { 'amountRemainingToPay' => 10 } } }
+      )]
+    end
     let(:patron) do
-      # create a patron with empty array for fines
-      instance_double(Symphony::Patron, barcode: '1', fines: [], can_pay_fines?: true, requests: [], checkouts: [],
-                                        remaining_checkouts: nil, barred?: false, status: 'OK', group?: false)
+      instance_double(Folio::Patron, barcode: '1', fines: fines, can_pay_fines?: true, requests: [],
+                                     checkouts: checkouts, remaining_checkouts: nil, barred?: false,
+                                     status: 'OK', group?: false)
     end
 
-    it 'does not show the shared computer payment alert' do
+    before do
+      assign(:fines, fines)
+      assign(:checkouts, checkouts)
+      without_partial_double_verification do
+        allow(view).to receive(:patron_or_group).and_return(patron)
+        allow(view).to receive(:patron).and_return(patron)
+        allow(fine).to receive(:to_partial_path).and_return('fines/fine')
+        allow(checkouts).to receive(:recalled?).and_return(false)
```

Comment:
Is there a reason we aren't verifying these doubles?

Reply:
Do you mean `without_partial_double_verification` ? The dumb answer is, without it the tests don't work and I get errors like
```
 1) fines/index with a FOLIO Fine shows the fined item author
     Failure/Error: allow(view).to receive(:patron_or_group).and_return(patron)
       #<ActionView::Base:0x0000000000a8c0> does not implement: patron_or_group
     # ./spec/views/fines/index.html.erb_spec.rb:95:in `block (3 levels) in <top (required)>'
     # /Users/marloelilongley/.gem/ruby/3.1.2/gems/webmock-3.18.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
```

If there is a different way to handle this I am open to it.

Reply:
@marlo-longley yes, I understand why you need that for `view`, but I don't see why `fine` or `checkouts` would give you that.

Reply:
Gotcha thanks for explaining. I just fixed this.